File: EaterSpring/src/main/kotlin/xyz/purema/eater/exceptions/ConflictException.kt
package xyz.purema.eater.exceptions

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import xyz.purema.eater.dtos.BaseResponseDto

class ConflictException(message: String) : AppException(message) {
    override fun getResponse(): ResponseEntity<BaseResponseDto> = ResponseEntity
        .status(HttpStatus.CONFLICT)
        .body(BaseResponseDto(message ?: "Unknown error"))
}

File: EaterNet/Eater/Exceptions/ConflictException.cs
using Eater.Dtos;
using Microsoft.AspNetCore.Mvc;

namespace Eater.Exceptions
{
    public class ConflictException : AppException
    {
        public ConflictException(string message) : base(message)
        {
        }

        public override IActionResult GetResponse()
        {
            return new ConflictObjectResult(new BaseResponseDto(Message));
        }
    }
}
